cmd/admin-handlers-idp-ldap.go

```diff
@@ -175,3 +178,359 @@ func (a adminAPIHandlers) AttachDetachPolicyLDAP(w http.ResponseWriter, r *http.
 
 	writeSuccessResponseJSON(w, encryptedData)
 }
+
+// AddServiceAccountLDAP adds a new service account for provided LDAP username or DN
+//
+// PUT /minio/admin/v3/idp/ldap/add-service-account
+func (a adminAPIHandlers) AddServiceAccountLDAP(w http.ResponseWriter, r *http.Request) {
```

Comment:
Most of the validations here are common with the AddServiceAccount handler and they are important - we should factor out the common parts into one or more smaller functions and reuse the code, so that if we fix something, it is automatically applied to the other API. Basically AddServiceAccountLDAP should be mostly identical to AddServiceAccount except for validating that the user exists in LDAP